analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Constraints/ObjectConstraint.cs

```diff
@@ -33,6 +39,9 @@ internal sealed class ObjectConstraint : SymbolicConstraint
         protected override string Name =>
             this == Null ? nameof(Null) : nameof(NotNull);
 
-        private ObjectConstraint() { }
+        private ObjectConstraint(ObjectConstraintKind kind) =>
+            Kind = kind;
```

Comment:
Constructor should be below the property

Reply:
I removed the num for now.

Reply:
I would keep it. It already helps.